src/com/sbezboro/standardplugin/listeners/PlayerTeleportListener.java

```diff
@@ -11,49 +13,55 @@
-		add(Material.DARK_OAK_FENCE);
-		add(Material.JUNGLE_FENCE);
-		add(Material.OAK_FENCE);
-		add(Material.SPRUCE_FENCE);
-		add(Material.NETHER_BRICK_FENCE);
-		add(Material.ACACIA_STAIRS);
-		add(Material.BIRCH_STAIRS);
-		add(Material.BRICK_STAIRS);
-		add(Material.COBBLESTONE_STAIRS);
-		add(Material.DARK_OAK_STAIRS);
-		add(Material.JUNGLE_STAIRS);
-		add(Material.NETHER_BRICK_STAIRS);
-		add(Material.QUARTZ_STAIRS);
-		add(Material.SANDSTONE_STAIRS);
-		add(Material.STONE_BRICK_STAIRS);
-		add(Material.SPRUCE_STAIRS);
-	}};
-
-	public PlayerTeleportListener(StandardPlugin plugin) {
-		super(plugin);
-	}
-	
-	@EventHandler(ignoreCancelled = true)
-	public void onPlayerTeleport(PlayerTeleportEvent event) {
-		Location to = event.getTo();
-
-		if (event.getCause() == TeleportCause.ENDER_PEARL) {
-			if (to.getWorld().getEnvironment() == Environment.NETHER &&
-					event.getTo().getBlockY() >= 124) {
-				event.setCancelled(true);
-				return;
-			}
-
-			if (blockedPearlMaterials.contains(to.getBlock().getType())) {
-				event.setCancelled(true);
-			}
-		}
-	}
+    private static final HashSet<Material> blockedPearlMaterials = new HashSet<Material>() {{
+        add(Material.IRON_BARS);
+        add(Material.ACACIA_FENCE);
+        add(Material.BIRCH_FENCE);
+        add(Material.DARK_OAK_FENCE);
+        add(Material.JUNGLE_FENCE);
+        add(Material.OAK_FENCE);
+        add(Material.SPRUCE_FENCE);
+        add(Material.NETHER_BRICK_FENCE);
+        add(Material.ACACIA_STAIRS);
+        add(Material.BIRCH_STAIRS);
+        add(Material.BRICK_STAIRS);
+        add(Material.COBBLESTONE_STAIRS);
+        add(Material.DARK_OAK_STAIRS);
+        add(Material.JUNGLE_STAIRS);
+        add(Material.NETHER_BRICK_STAIRS);
+        add(Material.QUARTZ_STAIRS);
+        add(Material.SANDSTONE_STAIRS);
+        add(Material.STONE_BRICK_STAIRS);
+        add(Material.SPRUCE_STAIRS);
+    }};
+
+    public PlayerTeleportListener(StandardPlugin plugin) {
+        super(plugin);
+    }
+
+    @EventHandler(ignoreCancelled = true)
+    public void onPlayerTeleport(PlayerTeleportEvent event) {
+        Location to = event.getTo();
+        StandardPlayer player = plugin.getStandardPlayer(event.getPlayer());
+        if (event.getCause() == TeleportCause.ENDER_PEARL) {
+            if (to.getWorld().getEnvironment() == Environment.NETHER &&
+                    event.getTo().getBlockY() >= 124) {
+                event.setCancelled(true);
+                return;
+            }
 
+            if (blockedPearlMaterials.contains(to.getBlock().getType())) {
+                event.setCancelled(true);
+            }
+        }
+        if (player.isPvpProtected() && event.getCause() == END_PORTAL) {
+            event.setCancelled(true);
+            player.sendMessage(ChatColor.DARK_PURPLE + "You cannot enter the end while PvP Protected! Use " +
+                    ChatColor.AQUA + "/enablepvp" + ChatColor.DARK_PURPLE + " to remove your PvP Protection!");
```

Comment:
why this change?

Reply:
Didn’t mean to click resolve convo on your yesterday’s comment, but I added if the player is not pvp protected to only log then, due to the adding of pvp protected players not being able to enter end so alts can’t abuse protection in end, I can remove both of those if you want